x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/create_rule_exceptions_route.ts

```diff
@@ -0,0 +1,279 @@
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+import type {
+  CreateRuleExceptionSchemaDecoded,
+  QueryRuleByIdSchemaDecoded,
+} from '../../../../../common/detection_engine/schemas/request';
+import {
+  createRuleExceptionsSchema,
+  queryRuleByIdSchema,
+} from '../../../../../common/detection_engine/schemas/request';
+import type { SecuritySolutionPluginRouter } from '../../../../types';
+import { DETECTION_ENGINE_RULES_URL } from '../../../../../common/constants';
+import { buildSiemResponse } from '../utils';
+import { patchRules } from '../../rules/patch_rules';
+import { buildRouteValidation } from '../../../../utils/build_validation/route_validation';
+import { readRules } from '../../rules/read_rules';
+import type { RuleParams } from '../../schemas/rule_schemas';
+
+export const createRuleExceptionsRoute = (router: SecuritySolutionPluginRouter) => {
+  router.post(
+    {
+      path: `${DETECTION_ENGINE_RULES_URL}/{id}/exceptions`,
+      validate: {
+        params: buildRouteValidation<typeof queryRuleByIdSchema, QueryRuleByIdSchemaDecoded>(
+          queryRuleByIdSchema
+        ),
+        body: buildRouteValidation<
+          typeof createRuleExceptionsSchema,
+          CreateRuleExceptionSchemaDecoded
+        >(createRuleExceptionsSchema),
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+
+      try {
+        const ctx = await context.resolve([
+          'core',
+          'securitySolution',
+          'alerting',
+          'licensing',
+          'lists',
+        ]);
+        const rulesClient = ctx.alerting.getRulesClient();
+        const listsClient = ctx.securitySolution.getExceptionListClient();
+
+        const { items } = request.body;
+        const { id: ruleId } = request.params;
+
+        // Check that the rule they're trying to add an exception list to exists
+        const rule = await readRules({
+          rulesClient,
+          ruleId: undefined,
+          id: ruleId,
+        });
+
+        if (rule == null) {
+          return siemResponse.error({
+            statusCode: 500,
+            body: `Unable to add exception to rule - rule with id:"${ruleId}" not found`,
+          });
+        }
+
+        let createdItems;
+
+        const ruleDefaultLists = rule.params.exceptionsList.filter(
+          (list) => list.type === ExceptionListTypeEnum.RULE_DEFAULT
+        );
+
+        // This should hopefully never happen, but could if we forget to add such a check to one
+        // of our routes allowing the user to update the rule to have more than one default list added
+        if (ruleDefaultLists.length > 1) {
+          return siemResponse.error({
+            statusCode: 500,
+            body: `Unable to add exception to rule - rule with id:"${ruleId}" has more than one default exception list assigned.`,
+          });
+        }
```

Comment:
Can this logic be replaced by `checkDefaultRuleExceptionListReferences` within a try/catch statement?

Reply:
Updated!